### PR TITLE
Add support for the SFTP protocol

### PR DESCRIPTION
**Pull request by bnfleb**
Support for SFTP protocol (issue #319)
I create a new client and a new fetcher to support SFTP.

### CODE REVIEW

File: commons/src/main/java/org/archive/net/ClientSFTP.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		this.logger.info("channelSftp put: " + paramString);
+		channelSFTP.put(paramFileInputStream, paramString);
+		return true;
+	}
+
+	protected boolean downRemoteSingleFile(String paramString1, String paramString2) throws Exception {
+		FileOutputStream fileOutputStream = new FileOutputStream(paramString2);
+		channelSFTP.get(paramString1, fileOutputStream);
+
+		this.logger.info("channelSftp download file: " + paramString1);
```
Comment (ato) [inline commons/src/main/java/org/archive/net/ClientSFTP.java:145]: Let's give these parameters meaningful names. I can overlook it when there's just one but `paramString2` is too much. ;-)

```suggestion
	protected boolean downRemoteSingleFile(String src, String dest) throws Exception {
		FileOutputStream fileOutputStream = new FileOutputStream(dest);
		channelSFTP.get(src, fileOutputStream);

		this.logger.info("channelSftp download file: " + src);
```

**Review by ato — CHANGES_REQUESTED** [review]
Sorry for taking so long to do this review.

This mostly looks okay but I noticed two serious problems (static session/channel etc fields in ClientSFTP and a byte loop index) and also have a couple of minor suggestions to improve the code readability.

**Comment by bnfleb** [comment]
Sorry for the old code and meaningless variable names, it came from a jar of heritrix1 but adapted for heritrix3.
It seems that the static fields are not necessary.

**Comment by ato** [comment]
Thanks!

**Pull request merged**